### PR TITLE
update spec & wpts

### PR DESCRIPTION
**Pull request by KhafraDev**

### CODE REVIEW

**Comment by KhafraDev** [comment]
Also getting a bunch of test failures with this message `TypeError [ERR_INVALID_STATE]: Invalid state: chunk ArrayBuffer is zero-length or detached"` (what's the best way of getting around this? none of my ideas are particularly good)

File: lib/fetch/dataURL.js [hunk 1]
```diff
@@ -126,7 +126,13 @@ function URLSerializer (url, excludeFragment = false) {
   const href = url.href
   const hashLength = url.hash.length
 
-  return hashLength === 0 ? href : href.substring(0, href.length - hashLength)
+  const serialized = hashLength === 0 ? href : href.substring(0, href.length - hashLength)
```
Comment (KhafraDev) [inline lib/fetch/dataURL.js:129]: this is a recent bug where serializing `https://localhost/#` would result in `https://localhost/#` instead of `https://localhost/`

**Comment by debadree25** [comment]
One of the cases to consider in bytes stream is to call ```c.byobRequest?.respond(0);``` after calling controller.close() you can take a look at https://github.com/nodejs/node/pull/49713/files maybe could be helpful?

File: lib/fetch/dataURL.js (hunk 1 above)
Comment (debadree25) [reply lib/fetch/dataURL.js:129]: I dont think endsWith would be slow any longer https://chromium-review.googlesource.com/c/v8/v8/+/4603612 :-)

Comment (tsctx) [reply lib/fetch/dataURL.js:129]: I forgot to add. I meant slower than the way I presented it.

**Comment by KhafraDev** [comment]
Thank you! That fixed most of the errors!

**Comment by KhafraDev** [comment]
We have one more error here:

```js
// 1. Let transformStream be a new TransformStream.
    // 2. Let identityTransformAlgorithm be an algorithm which, given chunk, enqueues chunk in transformStream.
    // 3. Set up transformStream with transformAlgorithm set to identityTransformAlgorithm and flushAlgorithm
    //    set to processResponseEndOfBody.
    const transformStream = new TransformStream({
      start () {},
      transform (chunk, controller) {
        controller.enqueue(chunk)
      },
      flush: processResponseEndOfBody
    })

    // 4. Set internalResponse’s body’s stream to the result of internalResponse’s body’s stream piped through transformStream.
    internalResponse.body.stream.pipeThrough(transformStream)
    internalResponse.body.stream = transformStream.readable
```

because the readable side of the transform stream isn't a byte stream... is there anything we could do about this? Or do we have to implement the equivalent of the transformstream?

File: lib/fetch/dataURL.js (hunk 1 above)
Comment (KhafraDev) [reply lib/fetch/dataURL.js:129]: I'm not really interested in optimizing against v8, and the link above mostly proves why

**Comment by KhafraDev** [comment]
as many as I needed to for byob support, I'm probably missing some but this is fast forwarding us a couple months

**Comment by ronag** [comment]
CI failing?

**Comment by KhafraDev** [comment]
windows 20: ```
 not ok 74 - test/redirect-request.js # time=1809.245ms

```

windows 18: ```
 not ok 74 - test/redirect-request.js # time=10253.305ms
```

**Comment by KhafraDev** [comment]
(not related to the WPTs, they all passed on ubuntu 18/20)
